src/Infrastructure/LeanCode.ConfigCat/ConfigCatInitializer.cs

```diff
@@ -0,0 +1,28 @@
+using ConfigCat.Client;
+using Microsoft.Extensions.Hosting;
+
+namespace LeanCode.ConfigCat;
+
+public sealed class ConfigCatInitializer : BackgroundService
+{
+    private readonly IConfigCatClient configCatClient;
+
+    public ConfigCatInitializer(IConfigCatClient configCatClient)
+    {
+        this.configCatClient = configCatClient;
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "?",
+        "CA1031:DoNotCatchGeneralExceptionTypes",
+        Justification = "We don't want any exceptions to be propagated."
+    )]
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        try
+        {
+            await configCatClient.GetAllKeysAsync(stoppingToken);
+        }
+        catch { }
```

Comment:
Logged, maybe?

Reply:
I'd expect the ConfigCat's client to take care of logging any failures here but sure, I can add something myself.

Reply:
Nono, if it is logged already, we don't need it to log again.

Reply:
Actually, hmm, looks like [all the exceptions except `OperationCanceledException` are swallowed anyway](https://github.com/configcat/.net-sdk/blob/57f08f27508e6f98c6e4db5ca305dfa4ade36b0f/src/ConfigCatClient/ConfigCatClient.cs#L395-L416). But yes, they will be logged through the adapter anyway.

Reply:
So, do we even need it or not? Last time I though we're dumping it.

Reply:
It will stay as an opt-in. Defaults provided in DI extensions don't register this but if you want to warm up the cache and be told earlier that something's wrong then you can use it.

Reply:
Oh, and the flavor based on `BackgroundService` shouldn't block the host from starting even if ConfigCat takes a long time to get ready.